### PR TITLE
feat: add optional support for TLS trusted_ca_keys extension

### PR DESCRIPTION
**Pull request by james-ctc**
feat: add documentation
fix: consistent type naming convention applied
fix: failed tests
feat: added signal to stop stop server
feat: SA_RESTART flag + notes on client certificates fix: memory leak
feat: testing non-blocking read
feat: patched test only includes the test that would fail
      on an unpatched OpenSSL
fix: logging to omit filename and line information fix: partial config support improved where trust anchors are missing

## Describe your changes

Added support for TLS trusted_ca_keys extension.
Note its use is disabled since changes to EVSE Security are needed to obtain the trust anchor certificates needed by the extension

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [X] I have made corresponding changes to the documentation
- [X] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: modules/EvseV2G/EvseV2G.cpp
Comment (corneliusclaussen) [inline modules/EvseV2G/EvseV2G.cpp:21]: Is this really neccessary? We may want to replace boost::log at some point, so the EVLOG_nn abstraction should be used I think. Maybe we can remove the funtion and line number from all warning logs instead as it makes logs quite unreadable

File: modules/EvseV2G/EvseV2G.cpp
Comment (corneliusclaussen) [inline modules/EvseV2G/EvseV2G.cpp:26]: see other comment about direct access to boost log

File: modules/EvseV2G/EvseV2G.cpp
Comment (james-ctc) [reply modules/EvseV2G/EvseV2G.cpp:21]: I don't have a good solution.
We should use a consistent logging approach.
I was asked to make the TLS server logging configurable rather than use the standard approach.

EVLOG_warning etc. all have a memory leak which makes use of valgrind/memcheck problematic (especially for unit tests).

With EVLOG_warning ... they will always show EvseV2G.cpp and that line number despite the error occurring elsewhere (staging/tls ...). 

The file & line number have been very useful in logs in the past to help track down an issue. But showing the wrong file and line isn't useful.

I will revert to EVLOG_warning << str; and EVLOG_error << str; since that is the least worst option.

File: modules/EvseV2G/EvseV2G.cpp
Comment (james-ctc) [reply modules/EvseV2G/EvseV2G.cpp:26]: see other comment's response

File: modules/EvseV2G/EvseV2G.cpp
Comment (corneliusclaussen) [reply modules/EvseV2G/EvseV2G.cpp:21]: Maybe we can change it such that info and warning don't output line numbers. Critical and error maybe still do. It would be nice to have a warning level that is not as verbose as it is now

Comment (james-ctc) [reply modules/EvseV2G/EvseV2G.cpp]: added a note to liblog so that removing file & line can be considered.
https://github.com/EVerest/liblog/issues/22

File: modules/EvseV2G/connection/tls_connection.cpp
Comment (SebaLukas) [inline modules/EvseV2G/connection/tls_connection.cpp:160]: You can use an reference `auto&` instead of `config.chains[0]`.
For example:
```
auto& config_chain_0 = config.chains.emplace_back();
config_chain_0.certificate_chain_file = cert_path.c_str();
[...]
```

**Review by SebaLukas — APPROVED** [review]
The changes in the EvseV2G module look good :+1: I just have one small comment.

**Comment by Pietfried** [comment]
Please wait for a review from @AssemblyJohn before merging

File: modules/EvseV2G/connection/tls_connection.cpp
Comment (james-ctc) [reply modules/EvseV2G/connection/tls_connection.cpp]: done

**Review by AssemblyJohn — APPROVED** [review]

**Comment by james-ctc** [comment]
> Please wait for a review from @AssemblyJohn before merging

@Pietfried review complete, awaiting code owner review/approval.

File: lib/staging/tls/extensions/helpers.cpp
Comment (barsnick) [inline lib/staging/tls/extensions/helpers.cpp:48]: Nit: I think you should set the stream properties only once, not for every element. (I'm not sure whether the compiler is capable of optimizing this, godbolt says it isn't):
```suggestion
    out << std::setw(2) << std::setfill('0') << std::hex;
    for (std::size_t i = 0; i < obj.size(); i++) {
        out << static_cast<std::uint32_t>(*ptr++);
    }
```
Same for other `operator<<()`.

Comment (james-ctc) [reply lib/staging/tls/extensions/helpers.cpp:48]: I tried that change and it doesn't work.
(the code is only used in unit tests anyway)

Before
```
[ RUN      ] DER.equal
010203040506070809
[       OK ] DER.equal (0 ms)
```
After
```
[ RUN      ] DER.equal
0123456789
[       OK ] DER.equal (0 ms)
```

**Review by Pietfried — APPROVED** [review]

**Pull request merged**
